src/directives/badge/badge.directive.ts

```diff
@@ -0,0 +1,170 @@
+import { Directive, ElementRef, Input, Renderer, OnInit, OnChanges, SimpleChanges } from '@angular/core';
+import { ColorService } from 'src/services/color/color.service';
+import { ThemeColor } from 'src/services/color/theme-color';
+import { ContrastService } from './../accessibility/contrast-ratio/contrast.service';
+
+export enum BadgeVerticalPosition {
+    Top = 'Top',
+    Bottom = 'Bottom',
+    Inline = 'Inline'
+}
+
+export enum BadgeHorizontalPosition {
+    Right = 'Right',
+    Left = 'Left',
+    Center = 'Center'
+}
+
+export enum BadgeSize {
+    Small = 'Small',
+    Medium = 'Medium',
+    Large = 'Large'
+}
+
+@Directive({
+    selector: '[uxBadge]'
+})
+export class BadgeDirective implements OnInit, OnChanges {
+    /**
+     * Define the badge background color
+     */
+    @Input() public set badgeBackgroundColor(color: string) {
+        this._backgroundColor = ThemeColor.parse(this._colorService.resolve(color));
+    }
+
```

Comment:
We include comments above all our inputs and outputs to allow the code editor to provide documentation to the consumer, eg: https://github.com/UXAspects/UXAspects/blob/develop-1.8.17/src/components/alert/alert.component.ts#L22